File: iosApp/CocktailCraft/Components/OrderCard.swift
import SwiftUI
import shared

struct OrderCard: View {
    let order: Order
    @Environment(\.isDarkMode) var isDarkMode
    
    var body: some View {
        VStack(alignment: .leading, spacing: AppTheme.Spacing.sm) {
            // Header with Order ID and Status
            HStack {
                Text("Order #\(order.id)")
                    .font(AppTheme.Typography.headline)
                    .fontWeight(.semibold)
                    .foregroundColor(AppColors.textPrimary(isDarkMode: isDarkMode))
                
                Spacer()
                
                OrderStatusBadge(status: order.status)
            }

            // Date
            Text("Date: \(order.date)")
                .font(AppTheme.Typography.subheadline)
                .foregroundColor(AppColors.textSecondary(isDarkMode: isDarkMode))

            // Items Section
            VStack(alignment: .leading, spacing: 4) {
                Text("Items:")
                    .font(AppTheme.Typography.subheadline)
                    .fontWeight(.medium)
                    .foregroundColor(AppColors.textPrimary(isDarkMode: isDarkMode))

                ForEach(order.items, id: \.name) { item in
                    OrderItemRow(item: item)
                }
            }

            Divider()

            // Total
            HStack {
                Text("Total:")
                    .font(AppTheme.Typography.subheadline)
                    .fontWeight(.semibold)
                    .foregroundColor(AppColors.textPrimary(isDarkMode: isDarkMode))
                
                Spacer()
                
                Text(order.total.asPrice)
                    .font(AppTheme.Typography.subheadline)
                    .fontWeight(.bold)
                    .foregroundColor(AppColors.textPrimary(isDarkMode: isDarkMode))
            }
        }
        .padding(AppTheme.Spacing.lg)
        .cardStyle()
    }
}

struct OrderStatusBadge: View {
    let status: String
    
    var body: some View {
        Text(status)
            .font(AppTheme.Typography.caption)
            .padding(.horizontal, AppTheme.Spacing.sm)
            .padding(.vertical, 4)
            .background(statusBackgroundColor)
            .foregroundColor(statusTextColor)
            .cornerRadius(AppTheme.CornerRadius.sm)
    }
    
    private var statusBackgroundColor: Color {
        switch status.lowercased() {
        case "processing":
            return AppColors.warning.opacity(0.2)
        case "delivered":
            return AppColors.success.opacity(0.2)
        case "cancelled":
            return AppColors.error.opacity(0.2)
        default:
            return Color.blue.opacity(0.2)
        }
    }
    
    private var statusTextColor: Color {
        switch status.lowercased() {
        case "processing":
            return AppColors.warning
        case "delivered":
            return AppColors.success
        case "cancelled":
            return AppColors.error
        default:
            return .blue
        }
    }
}

struct OrderItemRow: View {
    let item: OrderItem
    @Environment(\.isDarkMode) var isDarkMode
    
    var body: some View {
        HStack {
            Text("\(item.quantity)x \(item.name)")
                .font(AppTheme.Typography.caption)
                .foregroundColor(AppColors.textPrimary(isDarkMode: isDarkMode))
            
            Spacer()
            
            Text((item.price * Double(item.quantity)).asPrice)
                .font(AppTheme.Typography.caption)
                .fontWeight(.medium)
                .foregroundColor(AppColors.textPrimary(isDarkMode: isDarkMode))
        }
    }
}

#Preview {
    OrderCard(
        order: Order(
            id: "12345",
            date: "2023-12-01",
            status: "Processing",
            items: [
                OrderItem(name: "Classic Margarita", quantity: 2, price: 12.99),
                OrderItem(name: "Mojito", quantity: 1, price: 11.50)
            ],
            total: 37.48
        )
    )
    .environment(\.isDarkMode, false)
    .padding()
}

File: androidApp/src/main/java/com/cocktailcraft/android/ui/components/OrderCard.kt
package com.cocktailcraft.android.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.HorizontalDivider
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.cocktailcraft.android.R
import com.cocktailcraft.android.ui.theme.AppColors
import com.cocktailcraft.android.ui.theme.Spacing
import com.cocktailcraft.domain.model.Order

/**
 * Order summary card: order number + date header, line items, total, and a
 * colored status indicator. Extracted from OrderListScreen so the screen
 * stays a thin state/list shell.
 */
@Composable
fun OrderCard(
    order: Order,
    modifier: Modifier = Modifier
) {
    Card(
        modifier = modifier.fillMaxWidth(),
        shape = MaterialTheme.shapes.medium,
        colors = CardDefaults.cardColors(
            containerColor = AppColors.Surface
        ),
        elevation = CardDefaults.cardElevation(defaultElevation = 2.dp)
    ) {
        Column(
            modifier = Modifier.padding(Spacing.lg)
        ) {
            // Order header
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.CenterVertically
            ) {
                Text(
                    text = stringResource(R.string.order_number, order.id.takeLast(5)),
                    fontSize = 16.sp,
                    fontWeight = FontWeight.Bold,
                    color = AppColors.TextPrimary
                )

                Text(
                    text = order.date,
                    fontSize = 14.sp,
                    color = AppColors.TextSecondary
                )
            }

            Spacer(modifier = Modifier.height(Spacing.md))

            // Order items
            Column {
                order.items.forEach { item ->
                    Row(
                        modifier = Modifier
                            .fillMaxWidth()
                            .padding(vertical = Spacing.xs),
                        horizontalArrangement = Arrangement.SpaceBetween
                    ) {
                        Text(
                            text = stringResource(R.string.order_item_quantity_name, item.quantity, item.name),
                            fontSize = 14.sp,
                            color = AppColors.TextPrimary
                        )

                        Text(
                            text = stringResource(R.string.order_price_format, item.price * item.quantity),
                            fontSize = 14.sp,
                            fontWeight = FontWeight.Medium,
                            color = AppColors.TextPrimary
                        )
                    }
                }
            }

            HorizontalDivider(
                modifier = Modifier.padding(vertical = Spacing.md),
                color = AppColors.LightGray
            )

            // Total row
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.CenterVertically
            ) {
                Text(
                    text = stringResource(R.string.total),
                    fontSize = 16.sp,
                    fontWeight = FontWeight.Bold,
                    color = AppColors.TextPrimary
                )

                Text(
                    text = stringResource(R.string.order_price_format, order.total),
                    fontSize = 16.sp,
                    fontWeight = FontWeight.Bold,
                    color = AppColors.Primary
                )
            }

            Spacer(modifier = Modifier.height(Spacing.md))

            // Order status
            Row(
                verticalAlignment = Alignment.CenterVertically
            ) {
                val statusColor = when (order.status) {
                    "Completed" -> AppColors.Success
                    "In Progress" -> AppColors.Warning
                    else -> AppColors.Gray
                }

                Box(
                    modifier = Modifier
                        .size(10.dp)
                        .background(statusColor, CircleShape)
                )

                Spacer(modifier = Modifier.width(Spacing.sm))

                Text(
                    text = order.status,
                    fontSize = 14.sp,
                    color = statusColor
                )
            }
        }
    }
}
